FEXCore/Source/Interface/Core/ArchHelpers/Arm64Emitter.cpp

```diff
@@ -532,111 +719,241 @@ void Arm64Emitter::FillStaticRegs(bool FPRs, uint32_t GPRFillMask, uint32_t FPRF
+    const uint64_t SPOffset = AlignUp(GPRSize + FPRSize, 16);
+
+    // Spill static registers first.
+    SpillStaticRegs(TmpReg, true, x64::PreserveAll_SRAMask, CanUseSVE ? x64::PreserveAll_SRAFPRSVEMask : x64::PreserveAll_SRAFPRMask);
+
+    sub(ARMEmitter::Size::i64Bit, ARMEmitter::Reg::rsp, ARMEmitter::Reg::rsp, SPOffset);
+
+    // rsp capable move
+    add(ARMEmitter::Size::i64Bit, TmpReg, ARMEmitter::Reg::rsp, 0);
+
+    // Push the vector registers
+    PushVectorRegisters(TmpReg, CanUseSVE, DynamicFPRs);
+
+    // Push the general registers.
+    PushGeneralRegisters(TmpReg, DynamicGPRs);
+  }
+  else {
+    std::span<const FEXCore::ARMEmitter::Register> DynamicGPRs = x32::PreserveAll_Dynamic;
+    std::span<const FEXCore::ARMEmitter::VRegister> DynamicFPRs = x32::PreserveAll_DynamicFPR;
+
+    if (CanUseSVE) {
+      DynamicFPRs = x32::PreserveAll_DynamicFPRSVE;
+    }
+
+    const auto GPRSize = AlignUp(DynamicGPRs.size(), 2) * Core::CPUState::GPR_REG_SIZE;
+    const auto FPRSize = DynamicFPRs.size() * FPRRegSize;
+    const uint64_t SPOffset = AlignUp(GPRSize + FPRSize, 16);
+
+    // Spill static registers first.
+    SpillStaticRegs(TmpReg, true, x32::PreserveAll_SRAMask, CanUseSVE ? x32::PreserveAll_SRAFPRSVEMask : x32::PreserveAll_SRAFPRMask);
+
+    sub(ARMEmitter::Size::i64Bit, ARMEmitter::Reg::rsp, ARMEmitter::Reg::rsp, SPOffset);
+
+    // rsp capable move
+    add(ARMEmitter::Size::i64Bit, TmpReg, ARMEmitter::Reg::rsp, 0);
+
+    // Push the vector registers
+    PushVectorRegisters(TmpReg, CanUseSVE, DynamicFPRs);
+
+    // Push the general registers.
+    PushGeneralRegisters(TmpReg, DynamicGPRs);
+  }
+}
+
+void Arm64Emitter::FillForPreserveAllABICall(bool FPRs) {
+  const auto CanUseSVE = EmitterCTX->HostFeatures.SupportsAVX;
+
+  if (EmitterCTX->Config.Is64BitMode()) {
+    std::span<const FEXCore::ARMEmitter::Register> DynamicGPRs = x64::PreserveAll_Dynamic;
+    std::span<const FEXCore::ARMEmitter::VRegister> DynamicFPRs = x64::PreserveAll_DynamicFPR;
+
+    if (CanUseSVE) {
+      DynamicFPRs = x64::PreserveAll_DynamicFPRSVE;
+    }
+
+    // Fill static registers second.
+    FillStaticRegs(true, x64::PreserveAll_SRAMask, CanUseSVE ? x64::PreserveAll_SRAFPRSVEMask : x64::PreserveAll_SRAFPRMask);
+
+    // Pop vectors first
+    PopVectorRegisters(CanUseSVE, DynamicFPRs);
+
+    // Pop GPRs second
+    PopGeneralRegisters(DynamicGPRs);
+  }
+  else {
+    std::span<const FEXCore::ARMEmitter::Register> DynamicGPRs = x32::PreserveAll_Dynamic;
+    std::span<const FEXCore::ARMEmitter::VRegister> DynamicFPRs = x32::PreserveAll_DynamicFPR;
+
+    if (CanUseSVE) {
+      DynamicFPRs = x32::PreserveAll_DynamicFPRSVE;
+    }
+
+    // Fill static registers second.
+    FillStaticRegs(true, x32::PreserveAll_SRAMask, CanUseSVE ? x32::PreserveAll_SRAFPRSVEMask : x32::PreserveAll_SRAFPRMask);
+
+    // Pop vectors first
+    PopVectorRegisters(CanUseSVE, DynamicFPRs);
+
+    // Pop GPRs second
```

Comment:
ditto. also, the comments are out of sync with the code

Reply:
Deduplicated